Mathlib/LinearAlgebra/Basis/Cardinality.lean

```diff
@@ -38,36 +47,11 @@ variable [Semiring R] [AddCommMonoid M] [Nontrivial R] [Module R M]
 /--
 Over any nontrivial ring, the existence of a finite spanning set implies that any basis is finite.
 -/
-lemma basis_finite_of_finite_spans (w : Set M) (hw : w.Finite) (s : span R w = ⊤) {ι : Type w}
-    (b : Basis ι R M) : Finite ι := by
-  classical
-  haveI := hw.to_subtype
-  cases nonempty_fintype w
-  -- We'll work by contradiction, assuming `ι` is infinite.
-  rw [← not_infinite_iff_finite]
-  intro i
-  -- Let `S` be the union of the supports of `x ∈ w` expressed as linear combinations of `b`.
-  -- This is a finite set since `w` is finite.
-  let S : Finset ι := Finset.univ.sup fun x : w => (b.repr x).support
-  let bS : Set M := b '' S
-  have h : ∀ x ∈ w, x ∈ span R bS := by
-    intro x m
-    rw [← b.linearCombination_repr x, span_image_eq_map_linearCombination, Submodule.mem_map]
-    use b.repr x
-    simp only [and_true, eq_self_iff_true, Finsupp.mem_supported]
-    rw [Finset.coe_subset, ← Finset.le_iff_subset]
-    exact Finset.le_sup (f := fun x : w ↦ (b.repr ↑x).support) (Finset.mem_univ (⟨x, m⟩ : w))
-  -- Thus this finite subset of the basis elements spans the entire module.
-  have k : span R bS = ⊤ := eq_top_iff.2 (le_trans s.ge (span_le.2 h))
-  -- Now there is some `x : ι` not in `S`, since `ι` is infinite.
-  obtain ⟨x, nm⟩ := Infinite.exists_not_mem_finset S
-  -- However it must be in the span of the finite subset,
-  have k' : b x ∈ span R bS := by
-    rw [k]
-    exact mem_top
-  -- giving the desire contradiction.
-  simp only [self_mem_span_image, Finset.mem_coe, bS] at k'
-  exact nm k'
+lemma basis_finite_of_finite_spans [Nontrivial R] {s : Set M} (hs : s.Finite)
+    (hsspan : span R s = ⊤) {ι : Type w} (b : Basis ι R M) : Finite ι := by
+  have := congr(($hsspan).map b.repr)
+  rw [← span_image, Submodule.map_top, LinearEquivClass.range] at this
+  classical exact finite_of_span_finite_eq_top_finsupp (hs.image _) this
```

Comment:
```suggestion
  exact finite_of_span_finite_eq_top_finsupp (hs.image _) this
```

Reply:
done in [a379aac](https://github.com/leanprover-community/mathlib4/pull/23997/commits/a379aac351d685b1f8c401d113bb088b8bce247c)